core/Plugin/Report.php

```diff
@@ -847,17 +847,18 @@ public function getDimensions()
         if (!empty($this->getDimension())) {
             $dimensionId = str_replace('.', '_', $this->getDimension()->getId());
             $dimensions[$dimensionId] = $this->getDimension()->getName();
+            $tableDimensionId = $this->getDimension()->getId();
         }
 
-        if (!empty($this->getSubtableDimension())) {
-            $subDimensionId = str_replace('.', '_', $this->getSubtableDimension()->getId());
-            $dimensions[$subDimensionId] = $this->getSubtableDimension()->getName();
-        }
-
-        if (!empty($this->getThirdLeveltableDimension())) {
-            $subDimensionId = str_replace('.', '_', $this->getThirdLeveltableDimension()->getId());
-            $dimensions[$subDimensionId] = $this->getThirdLeveltableDimension()->getName();
-        }
+        for ($level = 1; $level <= 100; $level++) {
+            $subDimension = $this->getNthLevelTableDimension($level);
+            if (empty($subDimension) || $tableDimensionId === $subDimension->getId()) {
+                break;
+            }
+            $tableDimensionId = $subDimension->getId();
+            $subDimensionId = str_replace('.', '_', $subDimension->getId());
```

Comment:
```suggestion
            $subTableDimensionId = $subDimension->getId();
            $subDimensionId = str_replace('.', '_', $subTableDimensionId);
```

Reply:
We can't change that variable name, as it's used in the check above and is meant to be updated in each run. We though can reuse it.
```suggestion
            $tableDimensionId = $subDimension->getId();
            $subDimensionId = str_replace('.', '_', $tableDimensionId);
```